styx-scheduler-service/src/test/java/com/spotify/styx/docker/KubernetesDockerRunnerPodPollerTest.java

```diff
@@ -117,65 +109,48 @@ public void setUp() {
 
   @Test
   public void shouldSendRunErrorWhenPodForRunningWFIDoesntExist() {
-    Pod createdPod = createPod(WORKFLOW_INSTANCE, RUN_SPEC, SECRET_SPEC);
-    podList.setItems(Arrays.asList(createdPod));
-    createdPod.setStatus(podStatus1);
-    when(podStatus1.getPhase()).thenReturn("Pending");
-    when(k8sClient.pods().list()).thenReturn(podList);
-    when(namedPod2.get()).thenReturn(null);
-    when(k8sClient.pods().withName(POD_NAME_2)).thenReturn(namedPod2);
-    setupActiveInstances(RunState.State.RUNNING, POD_NAME, POD_NAME_2);
+    when(namedPod1.get()).thenReturn(null);
+    when(k8sClient.pods().withName(any())).thenReturn(namedPod1);
+    var states = setupActiveInstances(RunState.State.RUNNING, POD_NAME, POD_NAME_2);
+    var runState = states.get(WORKFLOW_INSTANCE);
 
-    kdr.tryPollPods();
+    kdr.poll(runState);
 
     verify(stateManager, times(1)).receiveIgnoreClosed(
-        Event.runError(WORKFLOW_INSTANCE_2, "No pod associated with this instance"), -1);
+        Event.runError(WORKFLOW_INSTANCE, "No pod associated with this instance"), -1);
   }
 
   @Test
   public void shouldNotSendRunErrorWhenPodForRunningWFIExists() {
-    Pod createdPod = createPod(WORKFLOW_INSTANCE, RUN_SPEC, SECRET_SPEC);
-    Pod createdPod2 = createPod(WORKFLOW_INSTANCE_2, RUN_SPEC, SECRET_SPEC);
+    var createdPod1 = createPod(WORKFLOW_INSTANCE, RUN_SPEC, SECRET_SPEC);
```

Comment:
But we still don't have `RUN_SPEC_1`. :trollface:

Reply:
Yeah the `TestData` thing should be cleaned up.